### ISSUE TITLE
Trouble installing dependencies

### ISSUE THREAD

**Issue by cmwaters** [tendermint/vue#118]
I'm trying to use the template package but when I try to run `npm install` I get the following long error log: https://gist.github.com/cmwaters/a9cedd3fc9bb92b33bf2595f4e1a5c14

Is anyone able to tell me what I might be doing wrong?

**Comment by fadeev** [comment]
Error in building native extensions for `node-sass`. Not sure what the exact error is yet.

**Comment by cmwaters** [comment]
Did you make any progress on the error. Could it be that I have something globally installed (not installed) that is affecting it?

**Comment by shapkarin** [comment]
# Just switch to the node v14.17.6

@cmwaters in past I got the same problem and find out that `node version` should be in sync with `node-sass version`. 
Here is the [Node version support policy](https://github.com/sass/node-sass#node-version-support-policy)

@starport/vue uses [node-sass ^4.12.0](https://github.com/tendermint/vue/blob/210bee8ef53fc92938682e687ff7ef30d1be5939/packages/vue/package.json#L41). And npm understand that like:
Use node-sass v4 with the latest minor updates – `v4.14.1` 

So you should use `node v14.*.*` for example the latest 14 is `v14.17.6` 
But you use [node v16.5.0](https://gist.github.com/cmwaters/a9cedd3fc9bb92b33bf2595f4e1a5c14#file-failed_install-log-L701) instead and I guess that's the main problem.

## So you can just switch to the `node v14.17.6` with `nvm`:
```
nvm install 14.17.6
nvm use 14.17.6
```

## As an additional
@cmwaters you don't need to set `node v14.17.6` as the default, instead you can automatically change yours version for the current directory. [lukechilds/zsh-nvm](https://github.com/lukechilds/zsh-nvm#auto-use) or [nodenv/nodenv](https://github.com/nodenv/nodenv#nodenv-local) and `.nvmrc` or `.node-version` files can help you with that.

----------




# Proposals
@fadeev I plan to provide the PR
I suggest to add package.json ["engines" field](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) and `.npmrc` – that setup will warn about node version compatibility. 
The last one should contain [engine-strict=true](https://docs.npmjs.com/cli/v7/using-npm/config#engine-strict) to force error throwing.

## And we will get the much `more informative error`:
```
% node -v
v16.5.0

% npm install
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for @starport/vue@0.1.52: 
wanted: {"node":"^14.0.0"} (current: {"node":"16.5.0","npm":"6.14.15"})
npm ERR! notsup Not compatible with your version of node/npm: @starport/vue@0.1.52
npm ERR! notsup Not compatible with your version of node/npm: @starport/vue@0.1.52
npm ERR! notsup Required: {"node":"^14.0.0"}
npm ERR! notsup Actual:   {"npm":"6.14.15","node":"16.5.0"}
```

### And pretty the same with `yarn`:
```
error @starport/vue@0.1.52: The engine "node" is incompatible with this module.
Expected version "^14.0.0". Got "16.5.0"
error Found incompatible module.
```

BTW to upgrade node-sass does not solve the problem, because it requires to update other dependencies as well.
So I will add the PR with the provided solution that outputs informative compatibility error.

**Comment by cmwaters** [comment]
Thanks @shapkarin!

**Comment by shapkarin** [comment]
@cmwaters I hope the my PR will improve DX and remove such problem. A few months ago I faced the same issue and I'm bet that not the only me and you got this obscure error.

**Comment by mojtaba-esk** [comment]
Using `node v14.17.6` was better but faced this error:

```
Failed to compile.

./src/scss/app.scss (./node_modules/css-loader/dist/cjs.js??ref--8-oneOf-3-1!./node_modules/postcss-loader/src??ref--8-oneOf-3-2!./node_modules/sass-loader/dist/cjs.js??ref--8-oneOf-3-3!./src/scss/app.scss)
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
TypeError: this.getOptions is not a function
    at Object.loader (/home/moji/tendermint/blog/vue/node_modules/sass-loader/dist/index.js:25:24)
```

After searching, I found out if I downgrade sass-loader to version `^10.2.0`, it works fine for me.

**Comment by shapkarin** [comment]
I think that all kind of dependencies should be updated, especially devDependencies. But you still can get the list of deprecated dependecies using @vue/cli@latest (4.5.13) and node@v16. devDependencies that are generated with the lasets vue and @vue/cli has the same problem as well. It may be not so necessary fro the production build, but makes yours dev environment dangerous or incompatible.
Maybe better is to use `vite` #91 or `nuxt`.

**Comment by shapkarin** [comment]
In my env @vue/cli beta is only one cli that works fine without any warnings

**Comment by shapkarin** [comment]
To change my environment a bit helped to use stable @vue/cli 4.5.13. And according to the current issue, I'm still think that it's nice to have more informative warnings about the stack compatibility.

**Comment by marinhoarthur** [comment]
@shapkarin thanks for the idea on engine-strict. 

Unfortunately we can not fix node to `^14.0.0` because it would break for folks running Apple Silicon chips (node [added](https://nodejs.org/en/blog/release/v16.0.0/#toolchain-and-compiler-upgrades) support for M1 chips on `^16.0.0`)

We definitely need to upgrade/simplify all of our dependencies. We'll start a thread on how to best achieve that so to improve the DX (maybe using engine-strict with a different version).

For now, I would suggest trying out the `develop` branch in its latest. @clockworkgr has recently upgraded many dependencies. It should have fixed that issue for you.
